src/ipc/ipc4/helper.c

```diff
@@ -360,6 +361,7 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	/* Pass IPC to target core if both modules has the same target core */
 	if (!cpu_is_me(source->ipc_config.core) && source->ipc_config.core == sink->ipc_config.core)
 		return ipc4_process_on_core(source->ipc_config.core, false);
+	is_shared = (source->ipc_config.core != sink->ipc_config.core);
```

Comment:
shouldn't more formal?
is_shared = (source->ipc_config.core != sink->ipc_config.core）? true: false;

Reply:
I believe it is quite clear, I would leave as is

Reply:
> shouldn't more formal? is_shared = (source->ipc_config.core != sink->ipc_config.core）? true: false;

@btian1 no. `!=` is a boolean operation, its result is `true` or `false`. What you're proposing is similar to
```
bool fn();

bool x = fn();
return (x == true) ? true : false;
```
instead of just
```
return fn();
```
I'd actually go a step further and remove parentheses above, but, well...